src/main/kotlin/reposfinder/logic/ReposStorage.kt

```diff
@@ -0,0 +1,168 @@
+package reposfinder.logic
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.SerializationFeature
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import reposfinder.utils.Logger
+import java.io.File
+import java.io.FileOutputStream
+import java.util.concurrent.ConcurrentLinkedQueue
+
+class ReposStorage(
+    private val urls: List<String>,
+    private val dumpDirPath: String,
+    private val dumpThreshold: Int = 50,
+    private val uselessFieldsSuffixes: List<String> = listOf("_url"),
+    private var logger: Logger? = null
+) {
+    companion object {
+        const val REPO_DELIMITER = "/"
+        const val EXPLAIN = "explain"
+        const val GOOD_REPOS_DIR = "good"
+        const val BAD_REPOS_DIR = "bad"
+        const val GOOD_REPOS_FILE = "good_repos.jsonl"
+        const val BAD_REPOS_FILE = "bad_repos.jsonl"
+        const val GOOD_INPUT_URLS = "good_input_urls.jsonl"
+        const val BAD_INPUT_URLS = "bad_input_urls.jsonl"
+        const val SPLIT_SIZE = 3
+        const val OWNER_POS = 2
+        const val NAME_POS = 1
+    }
+
+    private val objectMapper = jacksonObjectMapper()
+
+    private val goodSummaryDir = File(dumpDirPath + File.separator + GOOD_REPOS_DIR)
+    private val badSummaryDir = File(dumpDirPath + File.separator + BAD_REPOS_DIR)
+    private val goodReposFile = File(dumpDirPath + File.separator + GOOD_REPOS_FILE)
+    private val badReposFile = File(dumpDirPath + File.separator + BAD_REPOS_FILE)
+    private val goodUrlsFile = File(dumpDirPath + File.separator + GOOD_INPUT_URLS)
+    private val badUrlsFile = File(dumpDirPath + File.separator + BAD_INPUT_URLS)
```

Comment:
To avoid OS-specific behavior use `.resolve`
```
File(dumpDirPath).resolve(GOOD_REPOS_DIR)
```

Reply:
fixed